### PR TITLE
feat: chain-selectors version check in CI

### PR DESCRIPTION
**Pull request by erikburt**
## Motivation

See CCIP-2394.

## Solution

Workflow which compares the `chain-selectors` current version and latest version. If the versions differ, the workflow will fail.

## Testing
- Downgraded chain-selector to v1.0.15 and ran the check: https://github.com/smartcontractkit/ccip/actions/runs/9392408097/job/25866583934?pr=974

### CODE REVIEW

File: .github/workflows/chain-selectors-check.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          only-modules: true
+          go-version-file: "go.mod"
+
+      - name: Get chain-selectors version
+        id: get-chain-selectors-version
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          current_chain_selector_version="$(go list -m -f '{{.Version}}' github.com/smartcontractkit/chain-selectors)"
```
Comment (momentmaker) [inline .github/workflows/chain-selectors-check.yml]: nit: don't think the `" "` is necessary?

**Review by momentmaker — APPROVED** [review]

File: .github/workflows/chain-selectors-check.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: Chain Selectors Version Check
+
+on:
```
Comment (kalverra) [inline .github/workflows/chain-selectors-check.yml:3]: Maybe also include on tags?

File: .github/workflows/chain-selectors-check.yml (hunk 1 above)
Comment (erikburt) [reply .github/workflows/chain-selectors-check.yml]: Removed

File: .github/workflows/chain-selectors-check.yml (hunk 2 above)
Comment (erikburt) [reply .github/workflows/chain-selectors-check.yml:3]: Added for `v*` tags

**Review by kalverra — APPROVED** [review]

**Pull request merged**